base_layer/core/src/proof_of_work/monero_rx.rs

```diff
@@ -231,25 +231,343 @@ fn verify_root(monero_data: &MoneroData) -> Result<(), MergeMineError> {
     Ok(())
 }
 
-fn merged_mining_subfield(header: &BlockHeader) -> SubField {
-    let hash = header.merged_mining_hash();
-    let depth = 0;
-    SubField::MergeMining(VarInt(depth), Hash::from(from_slice(&hash)))
-}
-
 fn verify_header(header: &BlockHeader, monero_data: &MoneroData) -> Result<(), MergeMineError> {
-    if !(monero_data
-        .coinbase_tx
-        .prefix
-        .extra
-        .0
-        .contains(&merged_mining_subfield(header)))
-    {
+    let mut found = false;
+    for item in monero_data.coinbase_tx.prefix.extra.0.clone() {
+        match item {
+            SubField::MergeMining(x, y) => {
```

Comment:
Please name these more descriptively - according to the `SubField` docs: `SubField::MergeMining(depth, merkle_root)` - but maybe since we're using that field for the block header hash: `SubField::MergeMining(depth, hash)`.